src/org/benf/cfr/reader/bytecode/analysis/opgraph/Op02WithProcessedDataAndRefs.java

```diff
@@ -1473,13 +1473,36 @@ private Statement createStatement(final Method method, VariableFactory variableF
     }
 
     private Expression getLiteralConstantPoolEntry(Method m, ConstantPoolEntry cpe) {
-        if (cpe instanceof ConstantPoolEntryLiteral) {
-            return new Literal(TypedLiteral.getConstantPoolEntry(cp, cpe));
-        }
-        if (cpe instanceof ConstantPoolEntryDynamicInfo) {
-            return getDynamicLiteral(m, (ConstantPoolEntryDynamicInfo)cpe);
-        }
-        throw new ConfusedCFRException("Constant pool entry is neither literal or dynamic literal.");
+      if (cpe instanceof ConstantPoolEntryLiteral) {
+        return new Literal(TypedLiteral.getConstantPoolEntry(cp, cpe));
+      }
+      if (cpe instanceof ConstantPoolEntryDynamicInfo) {
+        return getDynamicLiteral(m, (ConstantPoolEntryDynamicInfo) cpe);
+      }
+      if (cpe instanceof ConstantPoolEntryMethodHandle) {
+        return constructMethodHandleExpression((ConstantPoolEntryMethodHandle) cpe);
+      }
+      if (cpe instanceof ConstantPoolEntryMethodType) {
+        return constructMethodTypeExpression((ConstantPoolEntryMethodType) cpe);
+      }
+      throw new ConfusedCFRException("Constant pool entry is neither literal, dynamic literal, method handle or method type.");
+    }
+
+    private static final InferredJavaType OBJECT_TYPE = new InferredJavaType(RawJavaType.NULL, InferredJavaType.Source.EXPRESSION);
```

Comment:
The purpose of the InferredJavaType (as distinct from the JavaTypeInstance) is to allow types to be changed after we have more information about them - i.e an inferred java type might be assumed to be bool, but used in a context where it's definitely int - rather than have massive action/subscriber chains, this allows the change to be visible in all sites where it's used.

Because of this, a static inferredjavatype is probably a bad idea - if this was used in one context, it would affect others.

There's a "locked" argument to IJT constructor which forbids this, but it's probably safer here to simply not use a static - will fix.

Reply:
Also we know this is returning a MethodType, so we should just use that. (done in https://github.com/leibnitz27/cfr/commit/1d0be9c9743dbb7e199d1f1ae1e45437881a4cb8 )